File: 0538-convert-bst-to-greater-tree/0538-convert-bst-to-greater-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    TreeNode prev = null;
    int cSum =0;
    public void convert(TreeNode node) {
        if(node == null) return;
        
        convert(node.right);
        cSum += node.val;
        node.val = cSum;
        prev = node;
        convert(node.left);
        
    }
    public TreeNode convertBST(TreeNode root) {
        convert(root);
        return root;
    }
}

File: 0538-convert-bst-to-greater-tree/0538-convert-bst-to-greater-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* prev=nullptr;
    int csum =0;
    void convert(TreeNode* node) {
        if(!node) return;
        
        convert(node->right);
        
        csum += node->val;
        node->val = csum;
        
        prev = node;
        convert(node->left);
    }
    TreeNode* convertBST(TreeNode* root) {
        convert(root);
        return root;
    }
};